### DISCUSSION TITLE
Task lists: allow to add a new task inline, without the need of editing the whole body

### DISCUSSION THREAD

**Discussion by gr2m** [community/community#4832] [Projects and Issues]
Once an issue body reaches a certain size, adding a new item at a certain place can be cumbersome. It would be great if I could add a new item to the end of a list. After I created it I can move it up to where I want the item to be, that part is already working great 👍🏼

**Comment by arrowtype** [comment]
This is pretty close to what I’ve been thinking for awhile – I generally love the experience of writing in markdown, and having that turn into rich text / images on submission. But, in some issues/PRs where I reach a point of complexity, I start to wish I could do more editing of checklists without jumping into markdown view entirely. 

For example, in [this issue](https://github.com/arrowtype/name-sans-roadmap/issues/37) I have about 64 checkboxes, and now I lose my place every time I go between editing and normal views. Usually, I just have it open in a browser while working in another app, then I jump back to check off items as I go – but of course, very often, I also realize I want to add or edit tasks, and I wish I could do that without losing my flow.

As @gr2m points out, it’s really nice that we can rearrange checkboxes in the normal view, so it seems like it might be feasible to add a way to edit them, too. I’m sure there are lots of good examples of this type of UX, but I think Dropbox Paper and Notion handle checkboxes in a pretty nice way.

As a longer-term thought, though, I can see a new question being raised by this: if checkboxes are directly editable, why wouldn’t the rest of issue comments also be? One answer could be: maybe instead of the current `read-only`|`editing` (rich text / markdown) views, there could instead be a `locked`|`unlocked` states. The `locked` state would be like the current read-only state, while in the `unlocked` state, the user could choose to edit in either `rich text`|`markdown` modes.
